src/libbson/tests/test-json.c

```diff
@@ -2727,6 +2727,7 @@ test_bson_json_errors (void)
        "Invalid read of boolean in state IN_BSON_TYPE"},
       {"{\"x\": {\"$dbPointer\": true}}",
        "Invalid read of boolean in state IN_BSON_TYPE_DBPOINTER_STARTMAP"},
+      {"[{\"$code\": {}}]", "Unexpected nested object value for \"$code\" key"},
```

Comment:
A test case to reject `{"$code": {}}`, which does not accept an object value. (Tests [CDRIVER-3380](https://jira.mongodb.org/browse/CDRIVER-3380))